stripe-core/src/test/java/com/stripe/android/core/model/serializers/CountryListSerializerTest.kt

```diff
@@ -0,0 +1,55 @@
+package com.stripe.android.core.model.serializers
+
+import com.google.common.truth.Truth.assertThat
+import com.stripe.android.core.model.Country
+import com.stripe.android.core.model.CountryCode
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.json.Json
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+
+@RunWith(RobolectricTestRunner::class)
+class CountryListSerializerTest {
+
+    private val json: Json = Json {
+        ignoreUnknownKeys = true
+        isLenient = true
+        encodeDefaults = true
+    }
+
+    @Test
+    fun testDeserialize() {
+        assertThat(
+            json.decodeFromString(
+                CountryListSerializer,
+                TEST_COUNTRIES_STRING
+            )
+        ).containsExactly(US, SG, BR)
+    }
+
+    @Test
+    fun testSerialize() {
+        assertThat(
+            json.encodeToJsonElement(CountryListSerializer, listOf(US, SG, BR)).toString()
+        ).isEqualTo(
+            TEST_COUNTRIES_STRING
+        )
+    }
+
+    private companion object {
+        private val US = Country(CountryCode("US"), "United States")
+        private val SG = Country(CountryCode("SG"), "Singapore")
+        private val BR = Country(CountryCode("BR"), "Brazil")
+
+        private val TEST_COUNTRIES_STRING = """
+            {"US":"United States","SG":"Singapore","BR":"Brazil"}
```

Comment:
Should we also test malformed input?

Reply:
Since this serializer is using a `mapSerialDescriptor<String, String>`, it will make sure the input string is a valid map. Any malformat input will be detected with error like `Expected start of the object '{', but had 'EOF' instead at path`

Reply:
@ccen-stripe it might still be good to verify this behavior. I won't block on it. But it does seem like a good idea.

Reply:
I think that would be testing the behavior of the descriptor but not the serilizer itself, will leave it as is :)